### ISSUE TITLE
Catapult Broker MongoDB Authentication

### ISSUE THREAD

**Issue by CrackTheCode016** [nemtech/catapult-server#78]
Hello,

Just wanted to report something that should probably be implemented down the line.  Whilst testing out a DB backup tool to backup chain data, the tool requires the mongoDB instance to have authentication (when you launch mongoDB with `--auth`, aka `SCRAM-SHA1`).  By default, however, broker is not configured to authenticate with the DB.  This is something that will both improve the security of the stack while also ensuring compatibility with such tools.

The primary change will be to add DB password and user fields to `config-database.properties`, have `DatabaseConfiguration.cpp` add those to the mongo connection for the broker process.  It's possible that an option `auth = true/false` could also be an option to enable or disable DB authentication.

Thanks,
Bader.

**Comment by Jaguar0625** [comment]
mongoDB allows user and password to be embedded in the database URI (see http://mongocxx.org/mongocxx-v3/configuration/). 

That maps to DatabaseConfiguration::DatabaseUri setting.

Since we want to allow connecting to unauthenticated mongo in certain scenarios (e.g. development), extra structure around credentials in DatabaseConfiguration is less helpful and more inflexible. 

TLDR: closing because connecting to authenticated mongo is currently possible and we want to still allow connections to unauthenticated mongo.
